### PR TITLE
Codemod: remove Section and Items imports if not used elsewhere in file

### PR DESCRIPTION
**Pull request by reidbarber**
Previously we weren't removing Section or Item imports, because they may still be used by other components. Now we check if the import is still used, and removed it if not.

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/WAI/ARIA/apg/)

### CODE REVIEW

File: packages/dev/codemods/src/s1-to-s2/src/codemods/codemod.ts [hunk 1]
```diff
@@ -203,9 +203,10 @@ export default function transformer(file: FileInfo, api: API, options: Options)
         t.isIdentifier(b.path.node.imported) &&
         (b.path.node.imported.name === 'Item' || b.path.node.imported.name === 'Section')
       ) {
-        // Keep Item and Section imports
-        // TODO: remove if they are unused
-        return;
+        // Keep Item and Section imports if they are still used
+        if (root.find(j.JSXIdentifier, {name: b.path.node.imported.name}).length > 0) {
```
Comment (devongovett) [inline packages/dev/codemods/src/s1-to-s2/src/codemods/codemod.ts:207]: Wouldn't this fail if you had another variable in the file that shadowed the declaration?

```js
import {Item} from '@adobe/react-spectrum';

function foo() {
  let Item = 'button';
  // this item shouldn't be replaced.
}
```

Probably should use the scope for this, e.g `b.referenced` should tell you if the binding is referenced. But I think you will need to do a `bindings[0]?.path.scope.crawl()` to re-update the scope. I tried this before but ran into an error.

Comment (reidbarber) [reply packages/dev/codemods/src/s1-to-s2/src/codemods/codemod.ts:207]: `b.referenced` didn't work, but `b.path.isReferencedIdentifier()` works better but doesn't totally cover us. I'll try to look deeper

Comment (reidbarber) [reply packages/dev/codemods/src/s1-to-s2/src/codemods/codemod.ts]: Should be fixed now, and added a new test

**Review by snowystinger — DISMISSED** [review]

**Review by snowystinger — DISMISSED** [review]

**Review by LFDanLu — APPROVED** [review]

**Review by snowystinger — APPROVED** [review]

File: packages/dev/codemods/src/s1-to-s2/__tests__/__snapshots__/menu.test.ts.snap [hunk 2]
```diff
@@ -68,7 +67,6 @@ const items = [
 
 exports[`Dynamic - Renames key to id 1`] = `
 "import { MenuItem, Menu, MenuTrigger, SubmenuTrigger, Button } from "@react-spectrum/s2";
-import { Item } from '@adobe/react-spectrum';
```
[the remark this answers is not in the document]
Comment (reidbarber) [reply packages/dev/codemods/src/s1-to-s2/__tests__/__snapshots__/menu.test.ts.snap:71]: Just tried this locally, and it kept the Item import. Are you seeing something different?

Before:
```jsx
import {Menu, Section, Item, MenuTrigger, Button} from '@adobe/react-spectrum';

let items = [
  {key: 'facebook', name: 'Facebook'},
  {key: 'twitter', name: 'Twitter'},
  {key: 'instagram', name: 'Instagram'},
].map(item => <Item key={item.key}>{item.name}</Item>);

<div>
  <MenuTrigger>
      <Button>Edit</Button>
      <Menu>
        <Section key={section.key} title="Static Menu Items">
          <Item key="cut">Cut</Item>
          <Item key="copy">Copy</Item>
        </Section> 
        <Section key={section.key} title="Social Media Sites">
          {items}
        </Section>
      </Menu>
    </MenuTrigger>
</div>
```

After:

```jsx
import { MenuSection, MenuItem, Menu, MenuTrigger, Button } from "@react-spectrum/s2";
import { Item } from '@adobe/react-spectrum';

let items = [
  {key: 'facebook', name: 'Facebook'},
  {key: 'twitter', name: 'Twitter'},
  {key: 'instagram', name: 'Instagram'},
].map(item => // TODO(S2-upgrade): Couldn't automatically detect what type of collection component this is rendered in. You'll need to update this manually.
<Item key={item.key}>{item.name}</Item>);

<div>
  <MenuTrigger>
      <Button>Edit</Button>
      <Menu>
        <MenuSection id={section.key}><Header>Static Menu Items</Header>
          <MenuItem id="cut">Cut</MenuItem>
          <MenuItem id="copy">Copy</MenuItem>
        </MenuSection> 
        <MenuSection id={section.key}><Header>Social Media Sites</Header>
          {items}
        </MenuSection>
      </Menu>
    </MenuTrigger>
</div>
```

[the remark this answers is not in the document]
Comment (ktabors) [reply packages/dev/codemods/src/s1-to-s2/__tests__/__snapshots__/menu.test.ts.snap:71]: I just wanted that in a test. :)

**Pull request merged**